hypervisor/src/mshv/mod.rs

```diff
@@ -244,7 +244,9 @@ impl hypervisor::Hypervisor for MshvHypervisor {
         }
 
         // Set additional partition property for SEV-SNP partition.
+        let mut _sev_snp_enabled = false;
```

Comment:
Interestingly, why you choose to not put this behind the same feature gate?

Reply:
That's mostly because rust does not allow hiding expression behind cfg attributes. https://github.com/rust-lang/rust/issues/15701

Reply:
Interesting. For this case, it believe it is simpler to do `sev_snp_enabled: mshv_vm_type == VmType::Snp,` and avoid such declaration. Please also rebase to resolve the clippy failures.

Reply:
Done.

Reply:
I believe you misunderstood what I meant:

```
        Ok(Arc::new(MshvVm {
            fd: vm_fd,
            msrs,
            dirty_log_slots: Arc::new(RwLock::new(HashMap::new())),
            #[cfg(feature = "sev_snp")]
            sev_snp_enabled: mshv_vm_type == VmType::Snp,
        }))
```